snuba/datasets/configuration/json_schema.py

```diff
@@ -119,22 +118,59 @@ def make_column_schema(
     "properties": {"key": TYPE_STRING, "set_key": TYPE_STRING},
 }
 
-QUERY_PROCESSORS_SCHEMA = {"type": "array", "items": TYPE_STRING}
+STORAGE_QUERY_PROCESSORS_SCHEMA = {"type": "array", "items": TYPE_STRING}
 
-KIND_SCHEMA = {"enum": ["writable_storage", "readable_storage"]}
+KIND_SCHEMA = {"enum": ["writable_storage", "readable_storage", "entity"]}
+
+ENTITY_QUERY_PROCESSOR = {
+    "type": "object",
+    "properties": {
+        "processor": TYPE_STRING,
+        "args": {"type": "object"},  # args are a flexible dict
+    },
+    "required": ["processor"],
+}
+
+ENTITY_VALIDATOR = {
+    "type": "object",
+    "properties": {
+        "validator": TYPE_STRING,
+        "args": {"type": "object"},  # args are a flexible dict
+    },
+    "required": ["validator"],
+}
+
+ENTITY_TRANSLATION_MAPPER_SUB_LIST = {
+    "type": "array",
+    "items": {
+        "type": "object",
+        "properties": {
+            "mapper": TYPE_STRING,
+            "args": {"type": "object"},
```

Comment:
The args for `functions` and `subscriptables` are not arbitrary. I think you can add them to the schema as well and ensure they are correctly populated.

Reply:
Unfortunately they are a little bit arbitrary. For example, for `functions`, the expected argument type is `Sequence[FunctionCallMapper]`, which has multiple concrete subclasses like `DefaultFunctionMapper`, `FunctionNameMapper`, `AggregateCurriedFunctionMapper`.